tests/test_edgar_fetch.py

```
@@ -0,0 +1,122 @@
+    out = edgar_fetch.fetch_master_index("20240131")
+    # assert the output is the expected content
+    assert out == b"OK"
+    # assert the call count is 3
+    assert call_count["n"] == 3
+
+    # test the fetch_master_index function retries then raises an error if the network call fails
+def test_fetch_master_index_retries_then_raises(monkeypatch):
+    calls = {"n": 0}
+
+    # define a fake get function that always raises a RequestException
+    def always_fail(_url, headers=None, timeout=None):
```

Ruff (0.14.8)

Unused function argument: `headers`

(ARG001)